AIToolbox/torchtrain/train_loop.py

```diff
@@ -121,6 +121,7 @@ def do_train(self, num_epoch, callbacks=None):
                 loss_batch.backward()
                 self.callbacks_handler.execute_gradient_update()
                 self.optimizer.step()
+                self.callbacks_handler.execute_optimizer_step()
```

Comment:
What are the performance implications of calling this all the time, even when not needed due to no callbacks being executed?

Reply:
Same goes for execute_gradient_update() above...